pkg/rbac/namespacedAuthorization.go

```diff
@@ -0,0 +1,90 @@
+// Copyright Contributors to the Open Cluster Management project
+package rbac
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"github.com/stolostron/search-v2-api/pkg/config"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/klog/v2"
+)
+
+type userData struct {
+	// impersonate kubernetes.Interface //client with impersonation config
+	err        error
+	namespaces []string
+	// resources  map[string][]string //key:namespace value list of resources
+	updatedAt time.Time
+	lock      sync.Mutex
+}
+
+func (cache *Cache) GetUserData(ctx context.Context, clientToken string) (*userData, error) {
+	var user userData
+	uid := cache.tokenReviews[clientToken].tokenReview.Status.User.UID //get uid from tokenreview
+	cache.userLock.Lock()
+	defer cache.userLock.Unlock()
+	cachedUserData, userDataExists := cache.users[uid] //check if userData cache for user already exists
+	if userDataExists {
+		klog.V(5).Info("Using user data from cache.")
+		return cachedUserData, cachedUserData.err
+
+	}
+	// create new instance
+	cache.users[uid] = &user
+	userData, err := user.getNamespaces(cache, ctx, clientToken)
+	return userData, err
+
+}
+
+func (user *userData) getNamespaces(cache *Cache, ctx context.Context, clientToken string) (*userData, error) {
```

Comment:
Logic in this function needs to be updated to get the namespaced-resources.
```
oc auth can-i --list -n <iterate-each-namespace>
```

Reply:
Unclear what the ask is here. Isn't this the authorization part?